### ISSUE TITLE
Parser doesn't support date time arithmetic in DATE_TRUNC arguments

### ISSUE THREAD

**Issue by mark-baole** [jOOQ/jOOQ#11987]
### SQL translator incorrectly translates this query:

```sql
SELECT
    CAST(date_column AS DATE) as report_date
    FROM
    (VALUES(SEQUENCE(DATE_TRUNC('MONTH',date('2021-05-01') - INTERVAL '1' MONTH),date('2021-05-01'),INTERVAL '1' DAY))) AS t1(date_array)
    CROSS JOIN UNNEST(date_array) AS t2(date_column)
```

### Parser output:

```sql
Token ')' expected: [4:60] ...ES(SEQUENCE(DATE_TRUNC('MONTH',date('2021-05-01') [*]- INTERVAL '1' MONTH),date('2021-05-01'),INTERVAL '1' DAY))) AS t1(date_array)
...
```

**Comment by lukaseder** [comment]
Thanks a lot for your report. Clearly, the subtraction should work.

I haven't seen that `SEQUENCE()` function yet, what does it do, and what SQL dialect is it from?

**Comment by lukaseder** [comment]
The date time arithmetic issue has been fixed for jOOQ 3.15.0, and will be deployed to https://www.jooq.org/translate, soon. I'll be happy to look into support for that `SEQUENCE()` function if it is from one of our supported dialects, in a separate issue.
